### PR TITLE
Add reset pins for halui Rapid Override, Feed Override and Spindle Overrides

### PR DESCRIPTION
**Pull request by russellgower**
Add reset pins for halui Rapid Override, Feed Override and Spindle Overrides

### CODE REVIEW

**Comment by petterreinholdtsen** [comment]
This commit should also update docs/man/man1/halui.1.

Would be nice if the commit message explained why it is needed.

**Comment by pippin88** [comment]
I am keen for this function.

Being able to quickly reset overrides would be very helpful.

Some GUIs implement their own pins to add this function - better to have the function in HALUI, not GUI dependent.

**Comment by petterreinholdtsen** [comment]
[pippin88]
> Some GUIs implement their own pins to add this function - better to
> have the function in HALUI, not GUI dependent.

Which GUIs implement their own pins for the reset function?  Do they
need to be patched to be compatible with this new halui functionallity?

-- 
Happy hacking
Petter Reinholdtsen

**Comment by russellgower** [comment]
I've added the missing man page update and have amended the commit message to give a bit of context. 

I found my git fu somewhat lacking when I tried to work out how to do that, but I think I got there in the end - hopefully I haven't messed it up to badly.

**Comment by petterreinholdtsen** [comment]
[russellgower]
> I've added the missing man page update and have amended the commit
> message to give a bit of context.

Look good.  I notice the first line of the commit message is a bit log
and has been cut.  Perhaps rephrase it to 'Add reset pins for halui
Rapid, Feed and Spindle Overrides' to make it fit within the github (and
git?) constraints?  Just nitpick.

> I found my git fu somewhat lacking when I tried to work out how to do
> that, but I think I got there in the end - hopefully I haven't messed
> it up to badly.

Look good to me.

You mentioned existing GUIs alrady having their own reset pins?  Which
are they?  No need to change them to use these new pins?

-- 
Happy hacking
Petter Reinholdtsen

**Comment by c-morley** [comment]
The HALUI program is just another GUI without the G. It has no higher priority or significance then the GUI you are using.
Now if your GUI is resetting something like an encoder count to 0 - you really should modify the GUI. 

Unfortunately there is no mechanism for linuxcnc's GUI/UIs to communicate to each others directly, nor any real protocol of how things would be preferred to work.

The encoder 'windup' problem sounds like a GMoccapy bug.

**Comment by zz912** [comment]
The HALUI program is just another GUI without the G. It has no higher priority or significance then the GUI you are using.
Now if your GUI is resetting something like an encoder count to 0 - you really should modify the GUI.

Unfortunately there is no mechanism for linuxcnc's GUI/UIs to communicate to each others directly, nor any real protocol of how things would be preferred to work.

The encoder 'windup' problem sounds like a GMoccapy bug.

**Comment by zz912** [comment]
I thought halui was superior to all GUIs.

I guess I'm describing my problem wrong. Gmoccapy override is intended for a potentiometer with mechanical stops that has a maximum and a minimum. I use the Quadratic encoder, which can take values from -infinity to +infinity.

If halui is not superior to Gmoccapy please ignore my previous post.

**Comment by c-morley** [comment]
Looking at the GMoccapy docs if you are using an encoder then you are probably using the wrong Gmoccapy pins. there are different pins for encoders vrs potentiometers.

**Comment by c-morley** [comment]
back to the code -  are these reset pins or preset pins?
